transports/webrtc/src/tokio/fingerprint.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use multihash::{Code, Hasher, Multihash, MultihashDigest};
+use webrtc::dtls_transport::dtls_fingerprint::RTCDtlsFingerprint;
+
+use std::fmt;
+
+const SHA256: &str = "sha-256";
+
+/// A certificate fingerprint that is assumed to be created using the SHA256 hash algorithm.
+#[derive(Eq, PartialEq, Copy, Clone)]
+pub struct Fingerprint([u8; 32]);
+
+impl Fingerprint {
+    pub(crate) const FF: Fingerprint = Fingerprint([0xFF; 32]);
+
+    #[cfg(test)]
+    pub fn raw(bytes: [u8; 32]) -> Self {
+        Self(bytes)
+    }
+
+    /// Creates a fingerprint from a raw certificate.
+    pub fn from_certificate(bytes: &[u8]) -> Self {
+        let mut h = multihash::Sha2_256::default();
+        h.update(bytes);
+
+        let mut bytes: [u8; 32] = [0; 32];
+        bytes.copy_from_slice(h.finalize());
+
+        Fingerprint(bytes)
+    }
+
+    /// Converts [`RTCDtlsFingerprint`] to [`Fingerprint`].
+    pub fn try_from_rtc_dtls(fp: &RTCDtlsFingerprint) -> Option<Self> {
+        if fp.algorithm != SHA256 {
+            return None;
+        }
+
+        let mut buf = [0; 32];
+        hex::decode_to_slice(fp.value.replace(':', ""), &mut buf).ok()?;
+
+        Some(Self(buf))
+    }
+
+    /// Converts [`type@Multihash`] to [`Fingerprint`].
+    pub fn try_from_multihash(hash: Multihash) -> Option<Self> {
+        if hash.code() != u64::from(Code::Sha2_256) {
+            // Only support SHA256 for now.
+            return None;
+        }
+
+        let bytes = hash.digest().try_into().ok()?;
+
+        Some(Self(bytes))
+    }
+
+    /// Converts this fingerprint to [`type@Multihash`].
+    pub fn to_multi_hash(self) -> Multihash {
```

Comment:
```suggestion
    pub fn to_multihash(self) -> Multihash {
```

To be consistent with the call above.

Reply:
Also this should be `into_multihash`, right?

https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

Reply:
Calling the function into_multihash would be misleading because the input is not consumed.

Reply:
> Also this should be `into_multihash`, right?
> 
> https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

`Fingerprint` is `Copy` :)